fix(metrics): ensure WAL metrics load after instance restart

GabriFedi97 opened the pull request:
This patch resolves an issue where WAL metrics were not available after a PostgreSQL instance restart until a configuration change occurred. The fix ensures that WAL metrics are correctly loaded upon startup.

Closes #6815

## Release notes

Fixed an issue where WAL metrics were not available after an instance restart until a configuration change was applied.

GabriFedi97 committed:
fix: always set instance ConfigSha256
The value of the instance.ConfigSha256 needs
to be set even when the pg config doesn't actually
change in the cluster spec. This is required in order
to have that value being populated on a first iteration
of the instance manager reconciliation, on a non-empty PGDATA.

This value is used to prevent some metrics from being calculated
multiple times when the pg configuration does not change,
but the fact that it was not populated the first time
prevented the calculation from being performed.

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

gbartolini commented:
@GabriFedi97 this is pointing to the wrong issue. Can you please fix the commit message?

mnencia commented:
@GabriFedi97 I think that the fix is correct, but I would also add a check in the metrics server https://github.com/cloudnative-pg/cloudnative-pg/blob/7aaa52d557c036e59a582c7e41b611c4e4f04d5c/pkg/management/postgres/webserver/metricserver/wal.go#L76 to load the values if they are missing regardless of the value of sha256.

GabriFedi97 committed:
fix: WAL metrics computation skip check
Modify the WAL metrics synchronizer to not
skip the values computation if the instance
config sha256 is empty. This helps the metrics
exporter to not return empty WAL values in case
the instance config sha256 has not been computed yet.

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

NiccoloFei committed:
chore: remove if in RefreshConfigurationFilesFromCluster

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei commented:
/test tl=4
